src/main/java/hudson/tasks/junit/TestResult.java

```diff
@@ -135,15 +135,15 @@ public TestResult(long buildTime, DirectoryScanner results, boolean keepLongStdi
      * Collect reports from the given {@link DirectoryScanner}, while
      * filtering out all files that were created before the given time.
      * @param keepLongStdio if true, retain a suite's complete stdout/stderr even if this is huge and the suite passed
-     * @param runId an optional {@link Run#getExternalizableId()} this test result was initially generated in.
      * @param nodeId an optional {@link org.jenkinsci.plugins.workflow.graph.FlowNode#id} for mapping between suite results and step executions.
      * @param enclosingBlocks Optional, possibly null list of enclosing {@link FlowNode#getId()}
-     * @since 1.21
+     * @param enclosingBlockNames Optional, possibly null list of enclosing block names
+     * @since 1.22
      */
-    public TestResult(long buildTime, DirectoryScanner results, boolean keepLongStdio, @CheckForNull String runId,
-                      @CheckForNull String nodeId, @CheckForNull List<String> enclosingBlocks) throws IOException {
+    public TestResult(long buildTime, DirectoryScanner results, boolean keepLongStdio, @CheckForNull String nodeId,
+                      @CheckForNull List<String> enclosingBlocks, @CheckForNull List<String> enclosingBlockNames) throws IOException {
```

Comment:
Same kind of issue here—feels like there needs to be some API type defining what information the parser needs, so we do not have to forever be doing a dance of deprecating constructors and methods and delegating to similar-looking overloads.